### PR TITLE
Automate unit test

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

### CODE REVIEW

File: wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java [hunk 1]
```diff
@@ -414,6 +415,16 @@ private void stopTestingServer() {
             } catch (Exception e) {
                 getLog().error("Error in closing the server", e);
             }
+        } else if (server.getServerType() != null && server.getServerType().equals(REMOTE_SERVER)) {
+            try {
+                if (getLog().isDebugEnabled()) {
+                    getLog().debug("Stopping docker container ");
+                }
+                Runtime.getRuntime().exec("docker stop wso2-mi-container").waitFor();
```
Comment (arunans23) [inline wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java:423]: Is the container name always fixed? We might have to rethink whether we are supporting only docker containers under REMOTE_SERVER.

Comment (kalaiyarasiganeshalingam) [reply wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java]: Undo these changes.

File: wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java [hunk 2]
```diff
@@ -721,40 +721,38 @@ private void printDetailedTable(List<List<String>> detailList, int columnSize, S
...
+            try {
+                FileUtils.copyDirectory(new File(Paths.get(projectRootPath, "deployment", "libs").
+                        toString()), new File(Paths.get(miDownloadPath.toString(),
+                        Constants.WSO2_MI_WITH_DASH + server.getServerVersion(), "lib").toString()));
+            } catch (IOException e) {
+                e.printStackTrace();
             }
+            String scriptPath = Paths.get(miDownloadPath.toString(), Constants.WSO2_MI_WITH_DASH +
+                    server.getServerVersion(), "bin", "micro-integrator.sh").toString();
+            Runtime.getRuntime().exec("chmod +x " + scriptPath);
```
Comment (SanojPunchihewa) [inline wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java:755]: Will this work in Windows as well?

File: wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java [hunk 3]
```diff
@@ -705,4 +718,118 @@ private void printDetailedTable(List<List<String>> detailList, int columnSize, S
...
+            server.setServerPath(scriptPath);
+        } catch (IOException e) {
+            getLog().error("Local server startup failed: " + e.getMessage(), e);
+        }
+    }
+
+    public File newFile(File destinationDir, ZipEntry zipEntry) throws IOException {
```
Comment (SanojPunchihewa) [inline wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java:762]: Can we have this as a private method?

File: wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java]: Updated with executable function.

**Review by SanojPunchihewa — APPROVED** [review]

**Review by arunans23 — APPROVED** [review]

**Pull request merged**